plugins/snapshot/src/main.ts

```diff
@@ -291,9 +293,12 @@ class SnapshotMixin extends SettingsMixinDeviceBase<Camera> implements Camera {
                 const cp = this.currentPicture;
                 debounced.catch(() => {});
                 try {
-                    picture = await timeoutPromise(1000, debounced);
+                    picture = await (options.timeout ? timeoutPromise(options.timeout, debounced) : debounced);
```

Comment:
regression here, as there is no timeout value provided. periodic snapshots must have a default timeout if none is provided. since it is periodic, erroring is fine, as the upstream will be updating the snapshot regularly and recover from errors. cache usage is also acceptable.

Reply:
@longzheng

Reply:
Ah, I wonder if the approach should be to have a default here

```ts
                    picture = await timeoutPromise(options.timeout ? options.timeout: 1000, debounced);
```

Or have `options` default `timeout` to `1000` elsewhere?

Reply:
Yep that's what I did. I dont think this should affect the initial issue of homekit having stale snapshots, but figured I'd bring it up. Change is already published.